src/Connector.AzureDataLake/Connector/AzureDataLakeConnector.cs

```diff
@@ -48,48 +78,229 @@ public override Task VerifyExistingContainer(ExecutionContext executionContext,
+            {
+                var syncItem = new SyncItem(streamModel.Id, connectorEntityData.EntityId, connectorEntityData.ChangeType, data, dataValueTypes);
+                await using var connection = new SqlConnection(configurations.StreamCacheConnectionString);
+                await connection.OpenAsync();
+                var tableName = GetCacheTableName(syncItem.StreamId);
+                await WriteToCacheTable(connection, syncItem, tableName);
             }
+            else
+            {
+                throw new NotSupportedException($"Buffer mode is not supported with '{StreamMode.EventStream}' mode.");
+            }
+
+            return SaveResult.Success;
+        }
+
+        private async Task WriteToCacheTable(
+            SqlConnection connection,
+            SyncItem syncItem,
+            string tableName)
+        {
+            var propertyKeys = syncItem.Data.Keys.Except(new[] { AzureDataLakeConstants.IdKey }).OrderBy(key => key).ToList();
 
-            if (connectorEntityData.IncomingEdges.SafeEnumerate().Any())
+            try
             {
-                data.Add("IncomingEdges", connectorEntityData.IncomingEdges);
+                await ExecuteWrite(connection, syncItem, tableName, propertyKeys);
+            }
+            catch (SqlException ex) when (GetIsTableNotFoundException(ex))
+            {
+                await EnsureCacheTableExists(connection, tableName, propertyKeys, syncItem.DataValueTypes);
+                await ExecuteWrite(connection, syncItem, tableName, propertyKeys);
+            }
+            catch(Exception ex)
+            {
+                _logger.LogError(ex, "Failed to process Entity with Id {EntityId}.", syncItem.EntityId);
+                throw;
             }
 
+            static async Task ExecuteWrite(SqlConnection connection, SyncItem syncItem, string tableName, List<string> propertyKeys)
+            {
+                if (syncItem.ChangeType == VersionChangeType.Removed)
+                {
+                    var deleteCommandText = $"DELETE FROM [{tableName}] WHERE {AzureDataLakeConstants.IdKey} = @{AzureDataLakeConstants.IdKey}";
+                    var command = new SqlCommand(deleteCommandText, connection)
+                    {
+                        CommandType = CommandType.Text
+                    };
+                    command.Parameters.Add(new SqlParameter($"@{AzureDataLakeConstants.IdKey}", syncItem.EntityId));
+                    var rowsAffected = await command.ExecuteNonQueryAsync();
+                    if (rowsAffected != 1)
+                    {
+                        throw new ApplicationException($"Rows affected for deletion is not 1, it is {rowsAffected}.");
+                    }
+                }
+                else
+                {
+                    var insertOrUpdateSql = $"""
+                        IF EXISTS (SELECT 1 FROM [{tableName}] WHERE {AzureDataLakeConstants.IdKey} = @{AzureDataLakeConstants.IdKey})  
```

Comment:
Potential race condition. Either 

rewrite as a `MERGE` 


or


use a transaction and add `WITH(XLOCK, ROWLOCK)` to the `SELECT`